models/taskRequests.js

```diff
@@ -0,0 +1,110 @@
+const { TASK_REQUEST_STATUS } = require("../constants/taskRequests");
+const { TASK_STATUS } = require("../constants/tasks");
+const firestore = require("../utils/firestore");
+const taskRequestsCollection = firestore.collection("taskRequests");
+const tasksModel = require("./tasks");
+const userModel = require("./users");
+
+/**
+ * Fetch all task requests
+ *
+ * @return {Object}
+ */
+const fetchTaskRequests = async () => {
+  const taskRequests = [];
+
+  const taskRequestsSnapshots = (await taskRequestsCollection.get()).docs;
+
+  for (const taskRequestsSnapshot of taskRequestsSnapshots) {
+    const taskRequestData = taskRequestsSnapshot.data();
+    const { requestors } = taskRequestData;
+
+    const { taskData } = await tasksModel.fetchTask(taskRequestData.taskId);
```

Comment:
You can move this line out of the for-loop by creating an array of promises(not writing await statements inside a for loop) and outside the for-loop this will be resolved with a `promise.all()`